File: python/163. Missing Ranges.py
import sys
class Solution(object):
    def findMissingRanges(self, nums, lower, upper):
        """
        :type nums: List[int]
        :type lower: int
        :type upper: int
        :rtype: List[str]
        """
        res = []
        nums = [lower - 1] + nums + [upper + 1]
        for i in range(len(nums) - 1):
            gap = nums[i+1] - nums[i]
            if gap != 1:
                if gap == 2:
                    res.append(str(nums[i] + 1))
                else:
                    res.append(str(nums[i] + 1) + "->" + str(nums[i+1] - 1))
        return res
                    
                
            
            

File: cpp/151-200/163. Missing Ranges.cc
class Solution {
public:
	string constructStr(long counter, long upper) {
		if (counter == upper - 1)
			return to_string(counter);
		else
			return to_string(counter) + "->" + to_string(upper - 1);
	}
	vector<string> findMissingRanges(vector<int>& nums, int lower, int upper) {
		vector<string> res;
		int length = nums.size();
		long prev = (long)lower;
		long cur;
		for (int i = 0; i <= length; i++) {
			if (i == length)
				cur = (long) upper + 1;
			else
				cur = (long) nums[i];
			if (prev < cur) {
				res.push_back(constructStr(prev, cur));
			}
			prev = cur + 1;
		}
		return res;
	}
};